### ISSUE TITLE
UnsupportedOperationException with Gradle 5.2.1 and Java 11 plugin

### ISSUE THREAD

**Issue by anuraaga** [gradle/gradle#8533]
### Expected Behavior

When using a plugin that has been published as Java 11 bytecode, Gradle 5.2 should build the project.

### Current Behavior

`UnsupportedOperationException` is thrown

```
Caused by: org.gradle.cache.CacheOpenException: Could not open cache directory 8mswczu3je5n44r8l0rmaev7x (C:\Users\Anuraag\.gradle\caches\5.2.1\gradle-kotlin-dsl-accessors\8mswczu3je5n44r8l0rmaev7x).
        at org.gradle.cache.internal.DefaultPersistentDirectoryStore.open(DefaultPersistentDirectoryStore.java:80)
        at org.gradle.cache.internal.DefaultPersistentDirectoryStore.open(DefaultPersistentDirectoryStore.java:42)
        at org.gradle.cache.internal.DefaultCacheFactory.doOpen(DefaultCacheFactory.java:94)
        at org.gradle.cache.internal.DefaultCacheFactory.open(DefaultCacheFactory.java:68)
        at org.gradle.cache.internal.DefaultCacheRepository$PersistentCacheBuilder.open(DefaultCacheRepository.java:118)
        at org.gradle.kotlin.dsl.cache.ScriptCache.cacheDirFor(ScriptCache.kt:67)
        at org.gradle.kotlin.dsl.cache.ScriptCache.cacheDirFor$default(ScriptCache.kt:52)
        at org.gradle.kotlin.dsl.accessors.AccessorsClassPathKt.cachedAccessorsClassPathFor(AccessorsClassPath.kt:99)
        at org.gradle.kotlin.dsl.accessors.AccessorsClassPathKt.buildAccessorsClassPathFor(AccessorsClassPath.kt:71)
        at org.gradle.kotlin.dsl.accessors.AccessorsClassPathKt.access$buildAccessorsClassPathFor(AccessorsClassPath.kt:1)
        at org.gradle.kotlin.dsl.accessors.AccessorsClassPathKt.projectAccessorsClassPath(AccessorsClassPath.kt:62)
        at org.gradle.kotlin.dsl.execution.Interpreter$ProgramHost.accessorsClassPathFor(Interpreter.kt:424)
        at Program.execute(Unknown Source)
        at org.gradle.kotlin.dsl.execution.Interpreter$ProgramHost.eval(Interpreter.kt:504)
        at org.gradle.kotlin.dsl.execution.Interpreter.eval(Interpreter.kt:182)
        at org.gradle.kotlin.dsl.provider.StandardKotlinScriptEvaluator.evaluate(KotlinScriptEvaluator.kt:114)
        at org.gradle.kotlin.dsl.provider.KotlinScriptPluginFactory$create$1.invoke(KotlinScriptPluginFactory.kt:51)
        at org.gradle.kotlin.dsl.provider.KotlinScriptPluginFactory$create$1.invoke(KotlinScriptPluginFactory.kt:36)
        at org.gradle.kotlin.dsl.provider.KotlinScriptPlugin.apply(KotlinScriptPlugin.kt:34)
        at org.gradle.configuration.BuildOperationScriptPlugin$1$1.run(BuildOperationScriptPlugin.java:69)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:402)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:394)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:165)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:250)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:158)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:92)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
        at org.gradle.configuration.BuildOperationScriptPlugin$1.execute(BuildOperationScriptPlugin.java:66)
        at org.gradle.configuration.BuildOperationScriptPlugin$1.execute(BuildOperationScriptPlugin.java:63)
        at org.gradle.configuration.internal.DefaultUserCodeApplicationContext.apply(DefaultUserCodeApplicationContext.java:48)
        at org.gradle.configuration.BuildOperationScriptPlugin.apply(BuildOperationScriptPlugin.java:63)
        at org.gradle.configuration.project.BuildScriptProcessor$1.run(BuildScriptProcessor.java:44)
        at org.gradle.internal.Factories$1.create(Factories.java:25)
        at org.gradle.api.internal.project.DefaultProjectStateRegistry$ProjectStateImpl.withMutableState(DefaultProjectStateRegistry.java:200)
        at org.gradle.api.internal.project.DefaultProjectStateRegistry$ProjectStateImpl.withMutableState(DefaultProjectStateRegistry.java:186)
        at org.gradle.configuration.project.BuildScriptProcessor.execute(BuildScriptProcessor.java:41)
        at org.gradle.configuration.project.BuildScriptProcessor.execute(BuildScriptProcessor.java:26)
        at org.gradle.configuration.project.ConfigureActionsProjectEvaluator.evaluate(ConfigureActionsProjectEvaluator.java:34)
        at org.gradle.configuration.project.LifecycleProjectEvaluator$EvaluateProject$1.run(LifecycleProjectEvaluator.java:106)
        ... 104 more
Caused by: java.lang.UnsupportedOperationException: This feature requires ASM7
        at org.jetbrains.org.objectweb.asm.ClassVisitor.visitNestMember(ClassVisitor.java:236)
        at org.jetbrains.org.objectweb.asm.ClassReader.accept(ClassReader.java:652)
        at org.jetbrains.org.objectweb.asm.ClassReader.accept(ClassReader.java:392)
        at org.gradle.kotlin.dsl.accessors.TypeAccessibilityProvider.invoke(AccessorsClassPath.kt:284)
        at org.gradle.kotlin.dsl.accessors.TypeAccessibilityProvider.kotlinVisibilityFor(AccessorsClassPath.kt:275)
        at org.gradle.kotlin.dsl.accessors.TypeAccessibilityProvider.isNonPublicKotlinType(AccessorsClassPath.kt:271)
        at org.gradle.kotlin.dsl.accessors.TypeAccessibilityProvider.loadAccessibilityInfoFor(AccessorsClassPath.kt:262)
        at org.gradle.kotlin.dsl.accessors.TypeAccessibilityProvider.access$loadAccessibilityInfoFor(AccessorsClassPath.kt:215)
        at org.gradle.kotlin.dsl.accessors.TypeAccessibilityProvider$accessibilityInfoFor$1.apply(AccessorsClassPath.kt:248)
        at org.gradle.kotlin.dsl.accessors.TypeAccessibilityProvider$accessibilityInfoFor$1.apply(AccessorsClassPath.kt:215)
        at org.gradle.kotlin.dsl.accessors.TypeAccessibilityProvider.accessibilityInfoFor(AccessorsClassPath.kt:247)
        at org.gradle.kotlin.dsl.accessors.TypeAccessibilityProvider.inaccessibilityReasonsFor(AccessorsClassPath.kt:239)
        at org.gradle.kotlin.dsl.accessors.TypeAccessibilityProvider.inaccessibilityReasonsFor(AccessorsClassPath.kt:232)
        at org.gradle.kotlin.dsl.accessors.TypeAccessibilityProvider.accessibilityForType(AccessorsClassPath.kt:225)
        at org.gradle.kotlin.dsl.accessors.AccessorsClassPathKt$availableProjectSchemaFor$1$1.invoke(AccessorsClassPath.kt:180)
        at org.gradle.kotlin.dsl.accessors.AccessorsClassPathKt$availableProjectSchemaFor$1$1.invoke(AccessorsClassPath.kt)
        at org.gradle.kotlin.dsl.accessors.ProjectSchemaEntry.map(ProjectSchemaProvider.kt:80)
        at org.gradle.kotlin.dsl.accessors.ProjectSchema.map(ProjectSchemaProvider.kt:57)
        at org.gradle.kotlin.dsl.accessors.AccessorsClassPathKt.availableProjectSchemaFor(AccessorsClassPath.kt:180)
        at org.gradle.kotlin.dsl.accessors.AccessorsClassPathKt.buildAccessorsFor(AccessorsClassPath.kt:151)
        at org.gradle.kotlin.dsl.accessors.AccessorsClassPathKt$buildAccessorsClassPathFor$$inlined$let$lambda$1.invoke(AccessorsClassPath.kt:73)
        at org.gradle.kotlin.dsl.accessors.AccessorsClassPathKt$buildAccessorsClassPathFor$$inlined$let$lambda$1.invoke(AccessorsClassPath.kt)
        at org.gradle.kotlin.dsl.accessors.AccessorsClassPathKt$cachedAccessorsClassPathFor$cacheDir$1.invoke(AccessorsClassPath.kt:102)
        at org.gradle.kotlin.dsl.accessors.AccessorsClassPathKt$cachedAccessorsClassPathFor$cacheDir$1.invoke(AccessorsClassPath.kt)
        at org.gradle.kotlin.dsl.cache.ScriptCache.initializeCacheDir(ScriptCache.kt:114)
        at org.gradle.kotlin.dsl.cache.ScriptCache.access$initializeCacheDir(ScriptCache.kt:37)
        at org.gradle.kotlin.dsl.cache.ScriptCache$cacheDirFor$1.execute(ScriptCache.kt:60)
        at org.gradle.kotlin.dsl.cache.ScriptCache$cacheDirFor$1.execute(ScriptCache.kt:37)
        at org.gradle.cache.internal.DefaultPersistentDirectoryCache$Initializer.initialize(DefaultPersistentDirectoryCache.java:98)
        at org.gradle.cache.internal.FixedSharedModeCrossProcessCacheAccess$1.run(FixedSharedModeCrossProcessCacheAccess.java:85)
        at org.gradle.cache.internal.DefaultFileLockManager$DefaultFileLock.doWriteAction(DefaultFileLockManager.java:207)
        at org.gradle.cache.internal.DefaultFileLockManager$DefaultFileLock.writeFile(DefaultFileLockManager.java:197)
        at org.gradle.cache.internal.FixedSharedModeCrossProcessCacheAccess.open(FixedSharedModeCrossProcessCacheAccess.java:83)
        at org.gradle.cache.internal.DefaultCacheAccess.open(DefaultCacheAccess.java:142)
        at org.gradle.cache.internal.DefaultPersistentDirectoryStore.open(DefaultPersistentDirectoryStore.java:78)
        ... 142 more

```

### Context

A project using the plugin worked on Gradle 5.1. When upgrading to Gradle 5.2, the build stopped working with the above exception.

Source code of plugin is [here](https://github.com/curioswitch/curiostack/blob/master/tools/gradle-plugins/gradle-golang-plugin/src/main/java/org/curioswitch/gradle/golang/GolangPlugin.java)

### Steps to Reproduce (for bugs)

```
$ git clone https://github.com/anuraaga/gradle-5.2-java11
$ cd gradle-5.2-java11
$ ./gradlew build --stacktrace
```

Requires JAVA_HOME to be openjdk11

### Your Environment

 * Build scan URL:  https://gradle.com/s/w3llzv5jckaye

**Comment by blindpirate** [comment]
Thanks for reporting.

@gradle/kotlin-dsl Seems like this is introduced by `kotlin DSL 1.1.3` -> `org.jetbrains.kotlin:kotlin-compiler-embeddable:1.2.51`. Are you aware of this issue?

**Comment by anuraaga** [comment]
Hello - just wondering if there's any update for this issue? Preventing upgrading to Gradle 5.2, which I'm very excited to do to try out the platform plugin :D

**Comment by bamboo** [comment]
Fixed by #8591.

**Comment by bamboo** [comment]
Thanks for the report, @anuraaga, fix will be in Gradle 5.3 RC1.
